p2p/net/conn/transport.go

```diff
@@ -0,0 +1,191 @@
+package conn
+
+import (
+	"net"
+
+	ma "github.com/ipfs/go-ipfs/Godeps/_workspace/src/github.com/jbenet/go-multiaddr"
+	manet "github.com/ipfs/go-ipfs/Godeps/_workspace/src/github.com/jbenet/go-multiaddr-net"
+	mautp "github.com/ipfs/go-ipfs/Godeps/_workspace/src/github.com/jbenet/go-multiaddr-net/utp"
+	reuseport "github.com/ipfs/go-ipfs/Godeps/_workspace/src/github.com/jbenet/go-reuseport"
+)
+
+type Transport interface {
+	manet.Listener
```

Comment:
a transport should not be a listener, it should have a `Listen()` function that returns a listener. a listener should not be opened until the user requests it to be opened.